api4/license.go

```diff
@@ -138,3 +141,59 @@ func removeLicense(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	ReturnStatusOK(w)
 }
+
+func requestTrialLicense(c *Context, w http.ResponseWriter, r *http.Request) {
+	auditRec := c.MakeAuditRecord("requestTrialLicense", audit.Fail)
+	defer c.LogAuditRec(auditRec)
+	c.LogAudit("attempt")
+
+	if !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_SYSTEM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+		return
+	}
+
+	if *c.App.Config().ExperimentalSettings.RestrictSystemAdmin {
+		c.Err = model.NewAppError("removeLicense", "api.restricted_system_admin", nil, "", http.StatusForbidden)
+		return
+	}
+
+	var usersNumber struct {
+		Users int `json:"users"`
+	}
+
+	b, readErr := ioutil.ReadAll(r.Body)
+	if readErr != nil {
+		c.Err = model.NewAppError("removeLicense", "api.license.request-trial.bad-request", nil, "", http.StatusBadRequest)
+		return
+	}
+	json.Unmarshal(b, &usersNumber)
+	if usersNumber.Users == 0 {
+		c.Err = model.NewAppError("removeLicense", "api.license.request-trial.bad-request", nil, "", http.StatusBadRequest)
+		return
```

Comment:
I've seen these blocks factored-out in model as smth. like usersNumber := model.GetUsersNumberFromJson(r.Body), I get that the type here is local/not reused, still personally like that approach better - 2/5

Reply:
Yes, probably makes sense, I'll do it :)